### ISSUE TITLE
Could only link 1 notes... bug.

### ISSUE THREAD

**Issue by ccchan234** [AlexW00/obsidian-note-linker#22]
**Describe the bug**

note contain 3 name of notes, suggested 3 links, but only linked 1 at last.

**To Reproduce**

in a new note, type in 3 already existing notes' name.

scan note, shows 3 but could only link 1.

**Expected behavior**

it showed 3, but after ticking, only show link 1.

**Screenshots**

note:
![image](https://user-images.githubusercontent.com/37549409/211603707-2f918943-e281-416a-bedb-87122141caca.png)

plugin:

3 ticks checked, bottom shows only 1.

![image](https://user-images.githubusercontent.com/37549409/211603868-1392bc58-3191-49e9-a624-390c4829f7f5.png)

**Desktop (please complete the following information):**
 - OS: [e.g. iOS] win10, 64bit
 - Version [e.g. 22] plugin v1.1.1; obsidian v 1.1. 9

**Smartphone (please complete the following information):**

**Additional context**

**Comment by ccchan234** [comment]
even i made a new vault,

install almost no other plugin,

it's same result:

![image](https://user-images.githubusercontent.com/37549409/211606445-6fd648d6-84b7-43e0-bb94-357e96863f46.png)

**Comment by ccchan234** [comment]
> even i made a new vault,
> 
> install almost no other plugin,
> 
> it's same result:
> 
> ![image](https://user-images.githubusercontent.com/37549409/211606445-6fd648d6-84b7-43e0-bb94-357e96863f46.png)

tested, even just have note linker as the only plugin, same result

**Comment by benccalcyxzfi** [comment]
I am also having the same problem

**Comment by Synteh** [comment]
Same problem here. Definitely a bug and not an individual issue.

**Comment by ccchan234** [comment]
most FOSS = abandoned ware, orphans... produced by producers for fun, without considerations of future

**Comment by Synteh** [comment]
> most FOSS = abandoned ware, orphans... produced by producers for fun, without considerations of future

Quite sad, since the idea is brilliant and no other plugin does what this does so well (in idea).
I have some coding knowledge, but not enough that in less than 30 minutes I could figure out what the problem seems to stem from. More time spent on it would not be worth it in my case, since in less then an hour, I used the current solution (of clicking them one by one) and I got what I wanted.

**Comment by schobernoise** [comment]
Same Problem here! What a Pity, it would have been the perfect plugin for me...

**Comment by ShortTimeNoSee** [comment]
Indeed, same problem. Here's to hoping.

**Comment by ccchan234** [comment]
> > most FOSS = abandoned ware, orphans... produced by producers for fun, without considerations of future
> 
> Quite sad, since the idea is brilliant and no other plugin does what this does so well (in idea). I have some coding knowledge, but not enough that in less than 30 minutes I could figure out what the problem seems to stem from. More time spent on it would not be worth it in my case, since in less then an hour, I used the current solution (of clicking them one by one) and I got what I wanted.

try gpt4.

i just used gpt4 to enhanced launchQ, a life changer for me i think.

i don't need this plugin anymore, otherwise i could help

**Comment by ccchan234** [comment]
i dont need this, because it links ALL keywords?

but now i agree /w obsidian's authors, that such linking is meaningless.

you should only link those thare are MEANINGFUL

so you could tag them on the Rt hand side one by one manually

**Comment by schobernoise** [comment]
> 

Well, this might be true, but it still is the case that you can choose which links will be created by ticking them... which saves a lot of time for linking obvious keywords. I have a lot of articles saved, which could be linked easily to important topics with this plugin.

**Comment by ccchan234** [comment]
> > 
> 
> Well, this might be true, but it still is the case that you can choose which links will be created by ticking them... which saves a lot of time for linking obvious keywords. I have a lot of articles saved, which could be linked easily to important topics with this plugin.

i think obsidian itself allow you to tag on the Right hand side. when it found unlinked but likely related thing, there is a "not linked" button there, you click it and they become linked.

**Comment by schobernoise** [comment]
Thank you, this works now for single notes, and also for a smaller amount of notes.
There is a problem though with linking a bigger amount (+2K). The plugin freezes and becomes unresponsive. Maybe a problem with loading all the links into memory?

**Comment by AlexW00** [comment]
> Thank you, this works now for single notes, and also for a smaller amount of notes. There is a problem though with linking a bigger amount (+2K). The plugin freezes and becomes unresponsive. Maybe a problem with loading all the links into memory?

Yeah, probably due to the removal of useMemo (which helps avoiding unnecessary re renders). If I have time I'll look into it again to improve performance.

**Comment by schobernoise** [comment]
Thank you very much, this is highly appreciated!
